### PR TITLE
Added disabled state to autocomplete and tweaked lookup's disabled state visually

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#99

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

File: e2e/autocomplete.e2e-spec.ts [hunk 1]
```diff
@@ -56,4 +56,19 @@ describe('Autocomplete', () => {
...
+    expect('#autocomplete-reactive').toMatchBaselineScreenshot(done, {
+      screenshotName: 'autocomplete-disabled'
+    });
+  });
+
+  it('should match previous screenshot while disabled (xs screen)', (done) => {
+    SkyHostBrowser.setWindowBreakpoint('xs');
+    element(by.css('#autocomplete-reactive-disable-button')).click();
+    expect('#autocomplete-reactive').toMatchBaselineScreenshot(done, {
+      screenshotName: 'autocomplete-disabled'
```
Comment (Blackbaud-AlexKingman) [inline e2e/autocomplete.e2e-spec.ts:71]: Should this be `autocomplete-disabled-xs` ?

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 2]
```diff
@@ -4,7 +4,8 @@ import {
   forwardRef,
   OnDestroy,
   OnInit,
-  Renderer2
+  Renderer2,
+  Input
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:8]: Check your alphabetical ordering here.

File: src/app/visual/autocomplete/autocomplete-visual.component.html [hunk 3]
```diff
@@ -51,6 +51,14 @@ <h2>
     </form>
   </div>
 
+  <button
+    id="autocomplete-reactive-disable-button"
+    class="sky-btn sky-btn-primary"
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/autocomplete/autocomplete-visual.component.html:56]: Check your alphabetical ordering here.

File: src/app/visual/autocomplete/autocomplete-visual.component.ts [hunk 4]
```diff
@@ -45,6 +49,20 @@ export class AutocompleteVisualComponent implements OnInit {
...
+    if (this.reactiveDisabledState) {
+      this.reactiveForm.get('favoriteColor').enable();
+    } else {
+      this.reactiveForm.get('favoriteColor').disable();
+    }
+
+    this.reactiveDisabledState = !this.reactiveDisabledState;
+  }
+
+  public toggleTemplateDisabled() {
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/autocomplete/autocomplete-visual.component.ts:62]: Methods need a return type.

File: e2e/autocomplete.e2e-spec.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply e2e/autocomplete.e2e-spec.ts:71]: Yup. Good call.

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:8]: Fixed

File: src/app/visual/autocomplete/autocomplete-visual.component.html (hunk 3 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/visual/autocomplete/autocomplete-visual.component.html:56]: Fixed. Also fixed a couple other places in this file that were wrong historically.

File: src/app/visual/autocomplete/autocomplete-visual.component.ts (hunk 4 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/visual/autocomplete/autocomplete-visual.component.ts:62]: Fix

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-AlexKingman addressed your comments. 

As for your question, that behavior matches what our other inputs do today when disabled and I verified that non-Sky inputs also are marked invalid when they are disabled.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts [hunk 5]
```diff
@@ -53,6 +54,20 @@ const SKY_AUTOCOMPLETE_VALIDATOR = {
 })
 export class SkyAutocompleteInputDirective implements OnInit, OnDestroy, ControlValueAccessor, Validator {
 
+  @Input()
```
Comment (blackbaud-johnly) [inline src/app/public/modules/autocomplete/autocomplete-input.directive.ts:57]: Should we go ahead and add a JSDoc comment here? I assume it just needs to say: `Indicates whether to disable the autocomplete field.` The autocomplete docs aren't updated to use JSDocs yet, but I think it's a good practice to add JSDoc comments as we go. I created https://github.com/blackbaud/skyux2-docs/issues/953 to document this new property in the existing docs and will handle that, but I think we should also add the JSDoc here as part of the dev work. Let me know if my suggested description doesn't work.

Comment (blackbaud-johnly) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:57]: I set https://github.com/blackbaud/skyux2-docs/pull/954 to update the docs.

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts (hunk 5 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:61]: Done

**Comment by blackbaud-ado** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable](https://host.nxt.blackbaud.com/skyux-lookup?_sp.spa=pr-autocomplete-disable)

File: src/app/public/modules/autocomplete/autocomplete-input.directive.ts (hunk 5 above)
Comment (blackbaud-johnly) [reply src/app/public/modules/autocomplete/autocomplete-input.directive.ts:61]: Looks good!

**Pull request merged**